server/src/main/java/io/crate/metadata/NodeContext.java

```diff
@@ -31,12 +33,23 @@ public class NodeContext {
     private final Functions functions;
     private final long serverStartTimeInMs;
     private final Roles roles;
+    private final Provider<Schemas> schemasProvider;
+
+    public static NodeContext withoutSchemas(Functions functions, Roles roles) {
```

Comment:
Thank you, continuing on your first approach https://github.com/crate/crate/pull/15988.

Reply:
I'd tend to agree with @mfussenegger. 
That we're currently using different `NodeContext` instances already is in my opinion already a bad design, and instead of moving this further I'd prefer to get rid of this. This looks cumbersome to me, even when protecting fishy calls by the exception.

Related to the suggestions of getting rid of the Schema injections. Wouldn't it even possible to get rid of injection in the related classes at all instead of moving the resolving from directly binding Schemas to resolving it over the bound NodeContext? Not sure if this is feasible (related to time spent on this feature), but as I think our overall goal is in this direction, maybe worth doing.

Reply:
Thanks for your input @seut , I actually needed to make more related classes away from injection in order to fix test failures for https://github.com/crate/crate/pull/15988. Please check the PR once it is ready as I am not sure if I understood your suggestion fully.